backend/deepchecks_monitoring/logic/check_logic.py

```diff
@@ -694,3 +697,60 @@ def reduce_check_window(model_results, monitor_options):
         if it is not None else None
         for model_version, it in model_results.items()
     }
+
+
+async def get_query_with_cache(
```

Comment:
maybe rename the function to something like `get_stored_query_result`?
because currently, its name does not help with understanding what it does